text/0000-file-source-map.md

```diff
@@ -0,0 +1,92 @@
+- Start Date: 2015-08-24
+- RFC PR: (leave this empty)
+- Refract Issue: (leave this empty)
+
+# Summary
+
+Update to Parse Result namespace Source Map element so it MAY carry information
+about the original source file.
+
+# Motivation
+
+Information about the original source file is needed in a situation where the
+final refract document with source maps is a result of processing multiple
+input files.
+
+This covers the situation where the final refract tree is created by parsing
+multiple API Blueprint files as described in
+[this issue](https://github.com/apiaryio/api-blueprint/issues/8).
+
+This RFC is SHOULD be considered before implementing the Parser Result namespace
+ as releasing it in its current form may lead into backward incompatible changes
+in the future.
+
+# Detailed design
+
+I suggest to remove the following from the current Parser Result namespace
+`Source Map` element:
+
+> The Source Map element MAY be used in any other element attributes in its
+normal ("unrefracted") form under the `sourceMap` property.
```

Comment:
Can `sourceMap` be both the unrefracted form and the full element form as listed below? Is this line restricting that?